samples/oauth_ccg_postman_pre_request_script/CCG Pre-request Script.postman_collection.json

```diff
@@ -0,0 +1,215 @@
+				},
+				"url": {
+					"raw": "http://kjur.github.io/jsrsasign/jsrsasign-latest-all-min.js",
+					"protocol": "http",
+					"host": [
+						"kjur",
+						"github",
+						"io"
+					],
+					"path": [
+						"jsrsasign",
+						"jsrsasign-latest-all-min.js"
+					]
+				}
+			},
+			"response": []
+		},
+		{
+			"name": "ccg-api-request",
+			"event": [
+				{
+					"listen": "prerequest",
+					"script": {
+						"exec": [
+							"const uuid = require('uuid')",
+							"",
+							"navigator = {}",
+							"window = {}",
+							"eval(pm.globals.get(\"jsrsasign-js\"))",
+							"",
+							"const getTextBetweenWords = (inputString, startWord, endWord) => {",
+							"    const regex = new RegExp(`${startWord}(.*?)${endWord}`);",
+							"    const match = inputString.match(regex);",
+							"    return match ? match[1].trim() : null;",
+							"}",
+							"",
+							"const setVariables = (aud) => {",
+							"    const authServer = getTextBetweenWords(aud, \"oauth2/\", \"/v\");",
+							"    pm.sendRequest(\"https://deptva-eval.okta.com/oauth2/\" + authServer + \"/.well-known/oauth-authorization-server\", function(err, response) {",
+							"        pm.globals.set(\"token_endpoint\", response.json().token_endpoint);",
+							"    })",
+							"}",
+							"",
+							"const getAssertion = () => {",
+							"    const secondsSinceEpoch = Math.round(Date.now() / 1000)",
+							"    const header = {",
+							"        alg: 'RS256',",
+							"        typ: 'JWT'",
+							"    }",
+							"    const claims = {",
+							"        \"aud\": pm.globals.get(\"aud\"),",
+							"        \"iss\": pm.globals.get(\"clientId\"),",
+							"        \"sub\": pm.globals.get(\"clientId\"),",
+							"        \"jti\": uuid.v4(),",
+							"        \"iat\": secondsSinceEpoch,",
+							"        \"exp\": secondsSinceEpoch + 60",
+							"    };",
+							"    const privateKey = pm.globals.get(\"privatePem\")",
+							"    const jwt = KJUR.jws.JWS.sign(null, header, claims, privateKey)",
+							"",
+							"    const urlencoded = [{",
+							"            key: 'grant_type',",
+							"            value: 'client_credentials',",
+							"            disabled: false",
+							"        },",
+							"        {",
+							"            key: 'client_assertion_type',",
+							"            value: 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer',",
+							"            disabled: false",
+							"        },",
+							"        {",
+							"            key: 'client_assertion',",
+							"            value: jwt,",
+							"            disabled: false",
+							"        },",
+							"        {",
+							"            key: 'scope',",
+							"            value: pm.globals.get(\"scope\"),",
+							"            disabled: false",
+							"        }",
+							"    ]",
+							"    return urlencoded",
+							"}",
+							"",
+							"const getToken = (urlencoded) => {",
+							"    const req = {",
+							"        url: pm.globals.get(\"token_endpoint\"),",
```

Comment:
Assuming some people will not read the readme lol, I would like to see some validation/messaging for the vars and log something like "X is not set but is required". I think currently if I do not set the AUD I get a type error which could result in an unecessary support request.